### DISCUSSION TITLE
Problem between optitrack and crazyswarm, rviz does not display cf‘s position, and the takeoff is in an endless loop

### DISCUSSION THREAD

**Discussion by Sherry-97** [USC-ACTLab/crazyswarm#411] [Q&A]
I am a beginner in crazyswarm, and I believe the problem I am facing now is the communication problem between crazy swarm and optitrack. I set the IP of optitrack (Windows) to **192.168.1.1** according to the requirements in the configuration, and set the ip to (**192.168.1.3**) under Ubuntu (**but I don’t know if my setting is correct, ipv4**). I’m using two The computer is connected to the switch with the optitrack camera through a network cable. And my hover configuration table. Very much hope for help。
![test rviz- - RViz_001](https://user-images.githubusercontent.com/70084772/123928213-49cc3500-d9c0-11eb-9ba8-ea476e2bbe75.png)
![12](https://user-images.githubusercontent.com/70084772/123928995-fa3a3900-d9c0-11eb-98d3-46f41818b229.png)
![Selection_006](https://user-images.githubusercontent.com/70084772/123928230-4df85280-d9c0-11eb-9769-3d5848cc0dfb.png)
![Editing Wired connection 1_004](https://user-images.githubusercontent.com/70084772/123928288-5c466e80-d9c0-11eb-80b1-1e9d094d7340.png)
<img width="122" alt="Snipaste_2021-06-30_16-12-33" src="https://user-images.githubusercontent.com/70084772/123928370-6ec0a800-d9c0-11eb-8b48-18a066a52923.png">
<img width="343" alt="Snipaste_2021-06-30_16-13-24" src="https://user-images.githubusercontent.com/70084772/123928385-72542f00-d9c0-11eb-8d95-223f47167567.png">

我按照之前问题中的建议尝试了https://github.com/whoenig/NatNetSDKCrossplatform，依然陷入死循环
![13](https://user-images.githubusercontent.com/70084772/123929046-09b98200-d9c1-11eb-9be3-21b3fb59c745.png)

**Comment by whoenig** [comment]
> The computer is connected to the switch with the optitrack camera through a network cable
You shouldn't do that. The OptiTrack switch is to be used *only* with the OptiTrack cameras. You should use a second Ethernet port to connect your PC to the Internet. The ideal setup is as follows:

```
Internet <-> Router <--+--- OptiTrack PC  --- OptiTrack Switch
                       |
                       +--- Ubuntu PC
```

The router usually runs a DHCP server, so you don't need to worry about assigning IPs manually. This way, you'll also get Internet on both the OptiTrack PC and the Ubuntu PC. You can use any standard router (with or without Wifi).

**Reply by Sherry-97** [reply]
Thank you very much for your quick reply，although my question is a bit amateur.
